include/executor/scheduler.hpp

```diff
@@ -12,10 +11,33 @@
 #include <vector>
 #include <type_traits>
 
+#define NO_COMPLETED_VALUE 99
+
 namespace Scheduler {
 
 typedef std::function<void(std::string, int, int, std::vector<std::string>)> ScheduleEventCallback;
 
+/**
+ * \note Scheduled Task Completion States
+ *
+ * SCHEDULED  - Has not yet run
+ * COMPLETED  - Ran successfully
+ * FAILED     - Ran once and failed
+ * RETRY_FAIL - Failed on retry
+ */
+namespace Completed {
+  static constexpr int SCHEDULED = 0;
+  static constexpr int SUCCESS = 1;
+  static constexpr int FAILED = 2;
+  static constexpr int RETRY_FAIL = 3;
```

Comment:
Do we also need "Retry_success"?

Reply:
We could use it for more nuanced record keeping, but I'm not sure if it's necessary. We more or less just needed to differentiate between tasks which may have failed for edge case scenarios vs tasks which most probably fail on every retry.